src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,49 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ELEM_NUMBER = 10;
+    private final K [] keys;
+    private final V [] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ELEM_NUMBER];
+        values = (V[]) new Object[MAX_ELEM_NUMBER];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getKeyIndex(key);
+        if (index != -1) {
+            values[index] = value;
+        } else {
+            keys[size] = key;
+            values[size] = value;
+            size++;
```

Comment:
You can do it like this
```suggestion
            values[size++] = value;
```